### DISCUSSION TITLE
New feature: insert elements from killring

### DISCUSSION THREAD

**Discussion by bagohart** [PatrickF1/fzf.fish#248] [Ideas]
Since 3.3, the content of the killring is visible via `$fish_killring`. (I guess it is NUL-separated, but it's not documented.)
Built-in behaviour is using  `ctrl+y` and `alt+y` to paste something from the killring into commandline, see https://fishshell.com/docs/current/interactive.html#copy-and-paste-kill-ring
I thought it would be nice to use fzf to choose something from it, so I hacked something together this morning, and it seems to work fine for me:
```fish
function fzf_select_from_killring
     set killring_entries (string split0 -- $fish_killring | fzf --multi)
     if test -n "$killring_entries"
         commandline -i (string join " " $killring_entries "") # adds a space after the inserted line output
     end
     commandline -f repaint
end
bind \e\ck fzf_select_from_killring
```
Are you interested in adding this feature to your plugin?

**Comment by PatrickF1** [comment]
Hi, sorry I am not at the moment. I'll let you know if I change my mind, thanks!

**Comment by bagohart** [comment]
Okay. In case anyone's interested in using this, I've packaged this functionality as a standalone plugin:
https://github.com/bagohart/fzf-killring.fish
It should be completely compatible to be used alongside fzf.fish, and if that ever changes, I will adapt it accordingly.

**Reply by PatrickF1** [reply]
Nice work! Looks great!
